src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs

```diff
@@ -162,6 +162,19 @@ public async Task<FormulaValue> EvalAsync(string expressionText, CancellationTok
             return await check.Expression.EvalAsync(parameters, cancel);
         }
 
+        public (IEnumerable<ExpressionError>, IEnumerable<FunctionInfo>) DefineFunctions(string script)

```

Comment:
Do you think we should wrap the members of this tuple up in a class with named properties?

Reply:
+1 that would be cleaner. 
And the perk is we can also add new properties without it being a breaking change.

Reply:
And maybe derive from IOperationStatus (like CheckResult does)